plugin.video.orftvthek/resources/lib/kodi.py

```diff
@@ -0,0 +1,420 @@
+        self.useragent = self.addon.getSetting('userAgent')
+        self.pager_limit = int(self.addon.getSetting('pagerLimit'))
+        self.max_cache_age = int(self.addon.getSetting('maxCacheAge')) * 60 * 60 * 24
+
+    def init_storage(self):
+        if not os.path.exists(self.data_folder):
+            os.makedirs(self.data_folder)
+
+    def translate(self, translation_id):
+        translation = self.addon.getLocalizedString
+        return translation(translation_id)
+
+    def get_translation(self, translation_id, fallback, replace=None):
+        translation = self.translate(translation_id)
+        if translation:
+            if replace is not None:
+                return replace % translation
+
+            return translation
+        return fallback
+
+    def is_geo_locked(self) -> bool:
+        return self.geo_lock
+
+    def hide_audio_description(self) -> bool:
+        return self.hide_audio_description_content
+
+    def hide_sign_language(self) -> bool:
+        return self.hide_sign_language_content
+
+    def hide_accessibility_menu(self) -> bool:
+        return self.hide_sign_language_content and self.hide_audio_description_content and not self.use_subtitles
+
+    def set_geo_lock(self, lock):
+        self.geo_lock = lock
+
+    def hide_content(self, item) -> bool:
+        if self.hide_audio_description_content and item.has_audio_description():
+            self.log("Hiding %s because AD content hide is enabled in settings" % item.label())
+            return True
+        if self.hide_sign_language_content and item.has_sign_language():
+            self.log("Hiding %s because OEGS content hide is enabled in settings" % item.label())
+            return True
+        if self.geo_lock and item.is_geo_locked():
+            self.log("Hiding %s because GEO Lock is active for your ISP" % item.label())
+            return True
+        return False
+
+    def render(self, item):
+        if not self.hide_content(item):
+            if item.is_playable():
+                list_item = self.render_video(item)
+                link = item.url()
+                route = self.plugin.url_for_path(link)
+                folder = self.use_segments and self.show_segments and item.has_segments()
+                addDirectoryItem(self.plugin.handle, url=route, listitem=list_item, isFolder=folder)
+            else:
+                list_item = self.render_directory(item)
+                link = item.url()
+                route = self.plugin.url_for_path(link)
+                addDirectoryItem(self.plugin.handle, url=route, listitem=list_item, isFolder=True)
+
+    def restart(self, video):
+        self.log("Running Restart Play Action")
+        play_item = self.render_video(video)
+        play_item.setProperty('inputstream.adaptive.play_timeshift_buffer', 'true')
+        streaming_url = video.get_stream().get('url')
+        Player().play(streaming_url, play_item)
+
```

Comment:
Please use the `setResolvedUrl` method to start playback.

Reply:
this call is used for playing the livestream from the start using the context menu. I couldn't find another way to do that since the context menu can't be playable (afaik). did this the same way in the previous versions as well (https://github.com/s0faking/plugin.video.orftvthek/blob/b76d8e2993a0d70a205b3073b14e3193d1e5ff8a/resources/lib/Addon.py#L233)